src/store/filter-presets.store.ts

```diff
@@ -0,0 +1,175 @@
+import { makeAutoObservable, reaction, toJS } from 'mobx'
+
+import { t } from '@i18n'
+import { BaseAsyncDataStore, TBaseDataStoreFetchOptions } from '@store/common'
+import {
+  ISolutionEntryDescription,
+  TCondition,
+} from '@service-providers/common'
+import { filteringProvider } from '@service-providers/filtering-regime'
+import { showToast } from '@utils/notifications/showToast'
+import { validatePresetName } from '@utils/validation/validatePresetName'
+import datasetStore from './dataset'
+
+type TFilterPresetStoreQuery = {
+  datasetName: string
+}
+
+class AvailablePresetsStore extends BaseAsyncDataStore<
```

Comment:
I think it's better to move async stores into separate files and give them. *.async.store.ts prefix, it would give a better understading

Reply:
fixed